src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -344,38 +339,35 @@
-    -->
     <ItemGroup>
-      <_BuiltArcadeSdkPackage Include="@(DependentRepoPackageFile)"
-                              Condition="!$([System.String]::new('%(Identity)').EndsWith('.symbols.nupkg')) and $([System.String]::new('%(Identity)').Contains($(_ArcadeSdkName)))" />
+      <SourceBuiltSdkOverride Include="@(_DependencyProducedPackage->WithMetadataValue('Identity', 'Microsoft.DotNet.Arcade.Sdk'))" Group="ARCADE" />
+      <SourceBuiltSdkOverride Include="@(_DependencyProducedPackage->WithMetadataValue('Identity', 'Microsoft.DotNet.SharedFramework.Sdk'))" Group="ARCADE_SHARED_FX_SDK" />
+      <SourceBuiltSdkOverride Include="@(_DependencyProducedPackage->WithMetadataValue('Identity', 'Microsoft.DotNet.CMake.Sdk'))" Group="ARCADE_CMAKE_SDK" />
     </ItemGroup>
+  </Target>
 
-    <PropertyGroup>
-      <_BuiltArcadeSdkPackageVersion>$([System.String]::new('%(_BuiltArcadeSdkPackage.FileName)').Substring($([MSBuild]::Add($(_ArcadeSdkName.Length), 1))))</_BuiltArcadeSdkPackageVersion>
-    </PropertyGroup>
+  <Target Name="DiscoverReferencePackageBuiltSdkOverrides"
+          DependsOnTargets="BuildRepoReferences;GetProducedPackagesFromTransitiveReferences"
+          Condition="'$(UseBootstrapSdks)' != 'true'">
+    <!--
+      Add Microsoft.Build.NoTargets and Microsoft.Build.Traversal SDK overrides, for all repos (except the one that builds this one, which opts out).
 
+      We need to override all SDK packages that are used by repos, to prevent NuGet SDK Resolver from trying to
+      restore these packages from online feeds. In offline build we only want to use local source-build feeds.
+      Therefore, we create a copy of repo's NuGet.config file, modify it, and pass it into repo build using
+      RestoreConfigFile property. MSBuild uses NuGetSdkResolver to restore missing SDKs, which doesn't honor
+      RestoreConfigFile property and uses NuGet.config file from repo's root which has online feeds.
+    -->
     <ItemGroup>
-      <SourceBuiltSdkOverride Include="Microsoft.DotNet.Arcade.Sdk" Group="ARCADE" Version="$(_BuiltArcadeSdkPackageVersion)" />
-      <SourceBuiltSdkOverride Include="Microsoft.DotNet.SharedFramework.Sdk" Group="ARCADE_SHARED_FX_SDK" Version="$(_BuiltArcadeSdkPackageVersion)" />
-      <SourceBuiltSdkOverride Include="Microsoft.DotNet.CMake.Sdk" Group="ARCADE_CMAKE_SDK" Version="$(_BuiltArcadeSdkPackageVersion)" />
-  
-      <SourceBuiltSdkOverride Include="@(ArcadeSdkOverride)" />
-      <SourceBuiltSdkOverride Include="@(ArcadeCMakeSdkOverride)" />
-      <SourceBuiltSdkOverride Include="@(ArcadeSharedFrameworkSdkOverride)" />
+      <SourceBuiltSdkOverride Include="@(_DependencyProducedPackage->WithMetadataValue('Identity', 'Microsoft.Build.Traversal'))" Group="TRAVERSAL" />
+      <SourceBuiltSdkOverride Include="@(_DependencyProducedPackage->WithMetadataValue('Identity', 'Microsoft.Build.NoTargets'))" Group="NOTARGETS" />
     </ItemGroup>
   </Target>
```

Comment:
Can we make this more flexible by only having on target which reads from registered msbuild sdks? We will eventually add more of these msbuild sdk entries and I don't think it's useful to have one target per repo origin.

Reply:
I'll see if I can make this more programatic.